docs/features/interceptors.md

```diff
@@ -66,29 +66,29 @@ Any "ordinary method" (i.e. with `MethodKind.Ordinary`) can have its calls inter
 
 File-local declarations of this type (`file class InterceptsLocationAttribute`) are valid and usages are recognized by the compiler when they are within the same file and compilation. A generator which needs to declare this attribute should use a file-local declaration to ensure it doesn't conflict with other generators that need to do the same thing.
 
-#### File paths
+In prior experimental releases of the feature, a well-known constructor signature `InterceptsLocation(string path, int line, int column)]` was also supported. Support for this constructor will be **dropped** prior to stable release of the feature.
 
-The *referenced syntax tree* of an `[InterceptsLocation]` is determined by normalizing the `filePath` argument value relative to the path of the containing syntax tree of the `[InterceptsLocation]` usage, similar to how paths in `#line` directives are normalized. Let this normalized path be called `normalizedInterceptorPath`. If exactly one syntax tree in the compilation has a normalized path which matches `normalizedInterceptorPath` by ordinal string comparison, that is the *referenced syntax tree*. Otherwise, an error occurs.
+#### Location encoding
 
-`#line` directives are not considered when determining the call referenced by an `[InterceptsLocation]` attribute. In other words, the file path, line and column numbers used in `[InterceptsLocation]` are expected to refer to *unmapped* source locations.
+The arguments to `[InterceptsLocation]` are:
+1. a version number. The compiler may introduce new encodings for the location in the future, with corresponding new version numbers.
+2. an opaque data string. This is not intended to be human-readable.
 
-Temporarily, for compatibility purposes, when the initial matching strategy outlined above fails to match any syntax trees, we will fall back to a "compat" matching strategy which works in the following way:
-- A *mapped path* of each syntax tree is determined by applying [`/pathmap`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.commandlinearguments.pathmap?view=roslyn-dotnet-4.7.0) substitution to `SyntaxTree.FilePath`.
-- For a given `[InterceptsLocation]` usage, the `filePath` argument value is compared to the *mapped path* of each syntax tree using ordinal string comparison. If exactly one syntax tree matches under this comparison, that is the *referenced syntax tree*. Otherwise, an error occurs.
-
-Support for the "compat" strategy will be dropped prior to stable release. Tracked by https://github.com/dotnet/roslyn/issues/72265.
+The "version 1" data encoding is a base64-encoded string consisting of the following data:
+- 16 byte xxHash128 content checksum of the file containing the intercepted call.
+- int32 in little-endian format for the one-based line number of the call in syntax.
+- int32 in little-endian format for the one-based column number of the call in syntax.
+- utf-8 string data containing a display file name, used for error reporting.
```

Comment:
This will have issues with encoding if you don't escape things. Especially with slashes and quotes. I recommend base64 this.